experiment/kind-conformance-image-e2e.sh

```diff
@@ -48,7 +39,7 @@ cleanup() {
 install_kind() {
     # install `kind` to tempdir
     TMP_DIR=$(mktemp -d)
-    curl -sLo "${TMP_DIR}/kind" https://github.com/kubernetes-sigs/kind/releases/download/${STABLE_KIND_VERSION}/kind-linux-amd64
+    curl -sLo "${TMP_DIR}/kind" https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64
```

Comment:
@BenTheElder wdyt with aligning with other CI jobs here? Alternatively, we can tag a new kind release.

Reply:
There was a new release tagged last night, but switching to latest here is reasonable.

I think this job should be demoted out of release blocking and presubmit given the lack of maintainership, it's frequently failing.
https://testgrid.k8s.io/conformance-all#conformance-image,%20master%20(dev)

For some additional context:

All of the other jobs using this `latest` binary the KIND maintainers actively support and we block kind PRs on running mirrors of the jobs. 

The `latest` (nightly / continuous) binary is only supported for upstream k/k testing where we need to move quicker to work around upstream breaking changes like this.

This image is really only consumed by sonobuoy, which is not upstream.

Reply:
Actually, it appears the CI version of this job is not even in release informing currently, let alone blocking.